### PR TITLE
chore: update MSRV to 1.74, update dependencies

### PR DESCRIPTION
**Pull request by mautamu**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. This is needed to update clap to ^4.5, which depends on rust >= 1.74.0.

## Type of change

- [X] Development change (no change visible to user)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## Updated user documentation:

Wiki needs MSRV updated from 1.70.0 to 1.74.0; README and CHANGELOG have been updated within this PR.

# Checklist:

- [X] Ran `make test` locally with no errors or warnings reported
  Note: To fully reproduce CI checks, you will need to run `make test-full-nix`. Usually, this is not necessary.
- [ ] Wiki pages have been updated accordingly (to perform **after** merge)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -430,8 +430,8 @@ A theme directory contains at least 2 files:
 These files need to be made executable. Many theme directories also contain:
 
 - `theme.ron`: which contains additional configuration options specific to the theme.
-- `polybar.ini`: a configuration file for the `polybar` application. *Need to have polybar installed!*
-- `picom.conf`: a configuration file for the `picom` compositor. *Need to have picom installed!*
+- `polybar.ini`: a configuration file for the `polybar` application. _Need to have polybar installed!_
+- `picom.conf`: a configuration file for the `picom` compositor. _Need to have picom installed!_
```
Comment (Eskaan) [inline README.md:434]: As you're already on it, how about
```suggestion
- `polybar.ini`: a configuration file for the `polybar` application. _You need to have polybar installed!_
- `picom.conf`: a configuration file for the `picom` compositor. _You need to have picom installed!_
```

Comment (mautamu) [reply README.md]: Thanks, and good catch as always @Eskaan! :)

**Review by Eskaan — APPROVED** [review]
I have read the changelog for [clap](https://github.com/clap-rs/clap/releases) and [nix](https://github.com/nix-rust/nix/blob/master/CHANGELOG.md), am I correct that the bumped msrv is due to the new clap version?

I have tested the built binaries to check if command parsing still works and it looks good so far. 

LGTM.

**Comment by mautamu** [comment]
Precisely @Eskaan! Apologies for not adding that to the documentation anywhere.

**Comment by Eskaan** [comment]
Thanks for doing the chores btw :heart:

**Pull request merged**